include/Logger.h

```diff
@@ -0,0 +1,126 @@
+// SPDX-FileCopyrightText: Deutsches Elektronen-Synchrotron DESY, MSK, ChimeraTK Project <chimeratk-support@desy.de>
+// SPDX-License-Identifier: LGPL-3.0-or-later
+#pragma once
+
+#include <ChimeraTK/cppext/future_queue.hpp>
+
+#include <boost/thread.hpp>
+
+#include <sstream>
+
+namespace ChimeraTK {
+
+  /********************************************************************************************************************/
+
+  class Logger {
+   public:
+    /**
+     * Obtain global instance of Logger singleton
+     */
+    static Logger& getInstance();
+
+    /**
+     * Severity levels used by the Logger. Note: there is no "fatal" severity, since the logger system does not work
+     * if the application is terminated immediately after sending a message to the log. Fatal errors shall be printed
+     * directly to std::cerr before terminating the application.
+     */
+    enum class Severity { trace, debug, info, warning, error };
+
+    /**
+     * Set the minimum severity level to be passed to the logger. By default, the minimum severity is set to
+     * Severity::info, so that trace and debug messages will not be processed. This will also prevent (to a certain
+     * extend) that the message text is composed at sender-side, which improves performance.
+     */
+    void setMinSeverity(Severity minSeverity) { _minSeverity = minSeverity; }
+
+    /**
+     * Proxy for output stream, handed out to the log sources by the Logger::Module.
+     */
+    class StreamProxy : public std::ostream {
+     public:
+      StreamProxy(Logger* logger, Severity severity, std::string context);
+      ~StreamProxy();
```

Comment:
Shouldn't ~StreamProxy() be virtual? It does override a virtual destructor. I'm getting a linter warning about it.

Reply:
Thanks, I will add the missing `override` keyword.